RELEASING.md

```diff
@@ -1,36 +1,41 @@
-# Making a rustls release
+## Before making a release
 
-This is a checklist for steps to make before/after making a rustls release.
-
-1. Attend to the README.md: this appears on crates.io for the release, and can't be edited after
-   the fact.
-   - Ensure the version has a good set of release notes.  Move old release notes to OLDCHANGES.md
-     if this is getting excessively long.
-   - Write the version and date of the release.
-2. Run `cargo update` followed by `cargo outdated`, to check if we have any
+1. Run `cargo update` followed by `cargo outdated`, to check if we have any
    dependency updates which are not already automatically taken by their semver specs.
    - If we do, take them if possible with separate commits (but there should've been
      dependabot PRs submitted for these already.)
-3. Now run `cargo test --all-features` to ensure our tests continue to pass with the
+2. Now run `cargo test --all-features` to ensure our tests continue to pass with the
    updated dependencies.
-4. Update `rustls/Cargo.toml` to set the correct version.
-5. Make a commit with the above changes, something like 'Prepare $VERSION'.  This
+3. Update `rustls/Cargo.toml` to set the correct version.
+4. Make a commit with the above changes, something like 'Prepare $VERSION'.  This
    should not contain functional changes: just versions numbers, and markdown changes.
-6. Do a dry run: in `rustls/` check `cargo publish --dry-run`
-7. Push the above commit.  Wait for CI to confirm it as green.
+5. Do a dry run: in `rustls/` check `cargo publish --dry-run`
```

Comment:
Maybe worth a small mention to make sure your worktree isn't dirty first? 

There's a warning from `cargo publish` if your worktree is dirty that mentions `--allow-dirty` but that's a trap: it won't _ignore_ the dirty portions of the worktree, it will include them in the release :skull_and_crossbones: I almost fell into that trap with a Python `venv` in the webpki release....

Reply:
added warnings about this